pkg/api/handlers/compat/resize.go

```diff
@@ -46,14 +46,6 @@ func ResizeTTY(w http.ResponseWriter, r *http.Request) {
 			utils.ContainerNotFound(w, name, err)
 			return
 		}
-		if state, err := ctnr.State(); err != nil {
-			utils.InternalServerError(w, errors.Wrapf(err, "cannot obtain container state"))
-			return
-		} else if state != define.ContainerStateRunning && !query.IgnoreNotRunning {
-			utils.Error(w, "Container not running", http.StatusConflict,
-				fmt.Errorf("container %q in wrong state %q", name, state.String()))
-			return
-		}
 		// If container is not running, ignore since this can be a race condition, and is expected
 		if err := ctnr.AttachResize(sz); err != nil {
 			if errors.Cause(err) != define.ErrCtrStateInvalid || !query.IgnoreNotRunning {
```

Comment:
I think we probably need to catch `define.ErrCtrStateInvalid` and return a `http.StatusConflict` down here. It should be a lot more forgiving than the above check, which only allows Running (we can also resize Created containers, and probably Paused containers as well, I forget) but ensures we're still returning good HTTP status codes.

Reply:
Not sure about the http status code, this is used for the compat api as well and the [docker api](https://docs.docker.com/engine/api/v1.41/#operation/ContainerResize) does not specify 409.

Reply:
fixed